test/Base_Tests/src/Semantic/Multi_Value_Convert_Spec.enso

```diff
@@ -57,6 +57,34 @@ add_specs suite_builder =
             Test.expect_panic Type_Error <| ac:U
             ac:V . v . should_equal 3.14
 
+        group_builder.specify "case of first" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc of
+                a:A -> a.a
+                b:B -> b.b
+                c:C -> c.c
+                _ -> "what?"
+            c1 . should_equal "a"
+
+        group_builder.specify "case of second downcasted" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc:B of
+                b:B -> b.b
+                a:A -> a.a
+                c:C -> c.c
+                _ -> "what?"
+            c1 . should_equal "b"
+
+        group_builder.specify "case of third downcasted" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc:C of
```

Comment:
Pattern matching on type on first value of a `EnsoMultiValue` works, but we want to match on other values as well:
```diff
diff --git test/Base_Tests/src/Semantic/Multi_Value_Convert_Spec.enso test/Base_Tests/src/Semantic/Multi_Value_Convert_Spec.enso
index ca768daaa0..2e54642532 100644
--- test/Base_Tests/src/Semantic/Multi_Value_Convert_Spec.enso
+++ test/Base_Tests/src/Semantic/Multi_Value_Convert_Spec.enso
@@ -84,6 +84,41 @@ add_specs suite_builder =
                 _ -> "what?"
             c1 . should_equal "c"
 
+        group_builder.specify "case of second" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc of
+                b:B -> b.b
+                a:A -> a.a
+                c:C -> c.c
+                _ -> "what?"
+            c1 . should_equal "b"
+
+        group_builder.specify "case of third" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc of
+                c:C -> c.c
+                b:B -> b.b
+                a:A -> a.a
+                _ -> "what?"
+            c1 . should_equal "c"
+
+        group_builder.specify "case of second from extra types" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc:A of
+                b:B -> b.b
+                a:A -> a.a
+                c:C -> c.c
+                _ -> "what?"
+            c1 . should_equal "b"
+
+        group_builder.specify "case of third  from extra types" <|
+            abc = 3.14 : A&B&C
+            c1 = case abc:A of
+                c:C -> c.c
+                b:B -> b.b
+                a:A -> a.a
+                _ -> "what?"
+            c1 . should_equal "c"
 
 main filter=Nothing =
     suite = Test.build suite_builder->
```

Reply:
- #11828

Reply:
These tests look great